app/code/core/Mage/Usa/Model/Shipping/Carrier/Ups/Source/Method.php

```diff
@@ -23,9 +23,20 @@ public function toOptionArray()
     {
         $ups = Mage::getSingleton('usa/shipping_carrier_ups');
         $arr = [];
+
```

Comment:
The only small issue here is that a lot of the elements in this array will be repeats, which is going to make it several times bigger than it needs to be. An easy way to fix that would be to use the 'value' as a key in `$arr`, and then take just the values without the keys, like this:

```
    public function toOptionArray()
    {
        $ups = Mage::getSingleton('usa/shipping_carrier_ups');
        $arr = [];

        // necessary after the add of Rest API
        $origins = $ups->getCode('originShipment');
        foreach ($origins as $origin) {
            foreach ($origin as $k => $v) {
                $arr[$k] = ['value' => $k, 'label' => Mage::helper('usa')->__($v)];
            }
        }

        // old XML API codes
        foreach ($ups->getCode('method') as $k => $v) {
            $arr[$k] = ['value' => $k, 'label' => Mage::helper('usa')->__($v)];
        }

        return array_values($arr);
    }
```

Reply:
yes but we're talking 20 elements instead of 10, I don't think it's something that needs to be optimized.

plus it will be removed later this year

I'll take a look at it later but

Reply:
I agree, it doesn't really matter. But, there's also the old saying "There's nothing more permanent than a temporary fix." :)

I went ahead and made an "approved" review. I don't feel strongly about this change, so it doesn't matter if you leave it as-is or tweak it.

Reply:
it is true, but I will personally remove it when the XML API will go down.

I'm all for cleaning the old mess and probably half of my PRs are cleanups heheheh